src/EditorFeatures/Test2/Rename/InlineRenameTests.vb

```diff
@@ -1902,5 +1907,33 @@ class [|$$Test1|]
                 Await VerifyTagsAreCorrect(workspace, "Test1")
             End Using
         End Function
+
+        <WpfTheory, WorkItem(44576, "https://github.com/dotnet/roslyn/issues/44576")>
+        <CombinatorialData, Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Async Function RenameFromOtherFile(host As TestHost) As Task
+            Using workspace = CreateWorkspaceWithWaiter(
+                <Workspace>
+                    <Project Language="C#" CommonReferences="true">
+                        <Document>
+                            class Test1
+                            {
+                                void Blah()
+                                {
+                                    [|$$Test2|] t2 = new [|Test2|]();
+                                }
+                            }
+                        </Document>
+                        <Document Name="Test2.cs">
+                            class Test2
+                            {
+                            }
+                        </Document>
+                    </Project>
+                </Workspace>, host)
+
+                Dim docToRename = workspace.Documents.First(Function(doc) doc.Name = "Test2.cs")
+                Await VerifyRenameOptionChangedSessionCommit(workspace, originalTextToRename:="Test2", renameTextPrefix:="Test2Changed", renameFile:=True, fileToRename:=docToRename.Id)
```

Comment:
add test with partial type since you check location count.

Reply:
Already covered with `VerifyNoFileRenameAllowedForPartialType`